internal/api/toolbox/list.go

```diff
@@ -122,29 +123,35 @@ func (t *toolbox[T, K, V]) listProjectWhenPermissionIsActivated(persesContext ap
 	// Last case, we want the list of the project that matches what the user has access to.
 	// So we get the list from the database, and then we keep only that one that matches the list extracted from the permission.
 	// The usage of the map is just to avoid having the o(n2) complexity by looping over two lists to make the intersection.
-	var result []any
 	projectList, listErr := t.metadataOrFullList(persesContext, parameters, query)
 	if listErr != nil {
 		return nil, listErr
 	}
+
 	switch typedList := projectList.(type) {
 	case []K:
+		result := make([]K, 0, len(typedList))
 		buildMap := buildMapFromList(typedList)
 		for _, project := range projects {
 			result = append(result, buildMap[project])
 		}
+		return result, nil
 	case []api.Entity:
+		result := make([]api.Entity, 0, len(typedList))
 		buildMap := buildMapFromList(typedList)
 		for _, project := range projects {
 			result = append(result, buildMap[project])
 		}
+		return result, nil
 	case [][]byte:
+		result := make([][]byte, 0, len(typedList))
 		buildMap := buildRawMapFromList(typedList)
 		for _, project := range projects {
 			result = append(result, buildMap[project])
```

Comment:
I haven't modified this line, but I am able to successfully retrieve the project 🤔

Reply:
It is difficult to implement something similar here at L109 because the results are used in the following sections, requiring additional branching for `[]interface{}`.
https://github.com/perses/perses/blob/v0.46.0/internal/api/toolbox/toolbox.go#L221-L223

At that time, it would also be necessary to create a new function like `func buildJSONBlobSlice(raws []interface{}) []byte`. I believe that JSON creation should be handled by a library (I also think that the buildJSON function should be removed in favor of using json.RawMessage).

https://github.com/perses/perses/blob/v0.46.0/internal/api/toolbox/toolbox.go#L46-L60

Reply:
mmm yeah ok.
Well the idea when implementing a custom json creation was to increase the performance when getting the data from the database. And the idea was to avoid un-marshalling data just for marshalling it again. That's where the custom json creation is coming from.
Now I'm wondering if we are using the json.RawMessage all the time if it will impact the performance or not.
If not then, perhaps it will make sense that the database is returning directly a `[]json.RawMessage` and not a `[][]byte`.
It should simplify a bit the code

Reply:
I can merge this PR, and if you want to make the refactoring, then I will be happy to review it :)